apps/browser/qml/pages/components/PopUpMenu.qml

```diff
@@ -117,22 +140,42 @@ SilicaControl {
                             flickable: menuFlickable
                         }
                     },
+                    Rectangle {
```

Comment:
With the new variable dimensions of `menuFlickable` I'm seeing some strange behaviour from the `VerticalScrollDecorator`. In practice it seems the values you need going in to the decorator are:
```
flickable.Height: menuFlickable.height - menuItem.menuTop - (2 * popUpMenu.margin)
flickable.contentHeight: menuFlickable.contentHeight - contentLoader.y
flickable.contentY: menuFlickable.contentY + menuItem.menuTop - contentLoader.y + (2 * popUpMenu.margin)
```
In other words, giving:
```
_sizeRatio: ((menuFlickable.height - menuItem.menuTop - (2 * popUpMenu.margin)) - _headerSpacing) / (menuFlickable.contentHeight - contentLoader.y + _topMenuSpacing + _bottomMenuSpacing)
y: Math.max(0, Math.min((parent.height / (menuFlickable.height - menuItem.menuTop - (2 * popUpMenu.margin))) * (_headerSpacing + ((menuFlickable.contentY + menuIt    em.menuTop - contentLoader.y + (2 * popUpMenu.margin)) - flickable.originY + _topMenuSpacing) * _sizeRatio), (parent.height - height)))
```

Reply:
Done, in a simplified fashion. The page header and other margins in the source item are things we explicitly don't want as the presence of a dialog header in a Page containing the PopUpMenu should not have any effect on the menu's scroll decorator.

Reply:
Okay, perfect, this is working nicely now. Thank you.